Sources/BSON/BSONDocument.swift

```diff
@@ -283,6 +283,13 @@ public struct BSONDocument {
                 if let doc = value.documentValue {
                     try BSONDocument.validate(doc.buffer)
                 }
+                if let array = value.arrayValue {
```

Comment:
since `BSON` is an enum, this looks like a good situation to use the `switch` syntax instead of repeated `if let`s